`azurerm_mssql_server_microsoft_support_auditing_policy` - New resource (#18609)
hashicorp/terraform-provider-azurerm: website/docs/r/mssql_server_microsoft_support_auditing_policy.html.markdown

  Manages a MS SQL Server Microsoft Support Auditing Policy.
---

# azurerm_mssql_server_microsoft_support_auditing_policy

Manages a MS SQL Server Microsoft Support Auditing Policy.

## Example Usage

```hcl
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_mssql_server" "example" {
  name                         = "example-sqlserver"
  resource_group_name          = azurerm_resource_group.example.name
  location                     = azurerm_resource_group.example.location
  version                      = "12.0"
  administrator_login          = "missadministrator"
  administrator_login_password = "AdminPassword123!"
}

resource "azurerm_storage_account" "example" {
  name                     = "examplesa"
  resource_group_name      = azurerm_resource_group.example.name
  location                 = azurerm_resource_group.example.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
}

resource "azurerm_mssql_server_microsoft_support_auditing_policy" "example" {
  server_id                  = azurerm_mssql_server.example.id

>   storage_endpoint           = azurerm_storage_account.example.primary_blob_endpoint

Author: ```suggestion
  blob_storage_endpoint      = azurerm_storage_account.example.primary_blob_endpoint
```

  resource_group_name = azurerm_resource_group.example.name
  server_name         = azurerm_mssql_server.example.name
  subnet_id           = azurerm_subnet.example.id

}

resource "azurerm_sql_firewall_rule" "example" {
  name                = "FirewallRule1"
  resource_group_name = azurerm_resource_group.example.name
  server_name         = azurerm_mssql_server.example.name
  start_ip_address    = "0.0.0.0"
  end_ip_address      = "0.0.0.0"
}

resource "azurerm_storage_account" "example" {
  name                = "examplesa"
  resource_group_name = azurerm_resource_group.example.name

  location                 = azurerm_resource_group.example.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
  account_kind             = "StorageV2"

  allow_nested_items_to_be_public = false

  network_rules {
    default_action             = "Deny"
    ip_rules                   = ["127.0.0.1"]
    virtual_network_subnet_ids = [azurerm_subnet.example.id]
    bypass                     = ["AzureServices"]
  }

  identity {
    type = "SystemAssigned"

  }
}

resource "azurerm_mssql_server_microsoft_support_auditing_policy" "example" {

>   storage_endpoint       = azurerm_storage_account.example.primary_blob_endpoint

Author: ```suggestion
  blob_storage_endpoint  = azurerm_storage_account.example.primary_blob_endpoint
```

  server_id              = azurerm_mssql_server.example.id
  log_monitoring_enabled = false

  storage_account_subscription_id = azurerm_subscription.primary.subscription_id

  depends_on = [
    azurerm_role_assignment.example,
    azurerm_storage_account.example,
  ]
}
```

## Arguments Reference

The following arguments are supported:

* `server_id` - (Required) The ID of the SQL Server to set the extended auditing policy. Changing this forces a new resource to be created.

* `enabled` - (Required) Whether to enable the extended auditing policy. Possible values are `true` and `false`. Defaults to `true`.

->**NOTE:**  If `enabled` is `true`, `blob_storage_endpoint` or `log_monitoring_enabled` are required.

> * `blob_storage_endpoint` - (Optional) The blob storage endpoint (e.g. https://example.blob.core.windows.net). This blob storage will hold all extended auditing logs.

Author: ```suggestion
* `blob_storage_endpoint` - (Optional) The blob storage endpoint (e.g. https://example.blob.core.windows.net). This blob storage will hold all Microsoft support auditing logs.
```
